# Live Code 3

**Name: Prabaningrum Hendraswari Buditjahjono**

**Batch:001- HCK**

## Description

### Problem Statement

Improving user/customer's experience in a service like travel insurance is important. Therefore, the company needs to find out what are the typical attributes in a person's travel that makes them claim their travel insurance so that they can determine what services would be fitting to offer to the public and minimize fraud.

## Working Area

```txt
Put your code, analysis, everything below this line

Make sure to check everything the rubric requires before you submit
```

In [175]:
import pandas as pd

In [176]:
import mysql.connector

db_con = mysql.connector.connect(
    host="remotemysql.com",
    user= "cESOsiaUiy",
    passwd= "saecUeCNAL",
    database= "cESOsiaUiy"
)

if db_con.is_connected():
    print("Successfully connected to MySQL database")

Successfully connected to MySQL database


In [177]:
#showing the dataset
df = pd.read_sql('SELECT * FROM travel_insurance', con=db_con)
df

c:\Users\praba\anaconda3\lib\site-packages\pandas\io\sql.py:761: UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
  warnings.warn(


id agency    agency_type distribution_channel        product_name  \
0          1    CBH  Travel Agency              Offline  Comprehensive Plan   
1          2    CBH  Travel Agency              Offline  Comprehensive Plan   
2          3    JZI       Airlines               Online          Value Plan   
3          4    JZI       Airlines               Online          Basic Plan   
4          5    KML  Travel Agency               Online        Premier Plan   
...      ...    ...            ...                  ...                 ...   
18214  18215    JZI       Airlines               Online          Basic Plan   
18215  18216    JZI       Airlines               Online          Basic Plan   
18216  18217    JZI       Airlines               Online          Basic Plan   
18217  18218    JZI       Airlines               Online          Basic Plan   
18218  18219    JZI       Airlines               Online          Basic Plan   

      claim  duration    destination  net_sales  commission gender  age  
0        No       186       MALAYSIA        -29        9.57      F   81  
1        No       186       MALAYSIA        -29        9.57      F   71  
2        No        66  UNITED STATES       -121       42.35      F   44  
3        No         1       MALAYSIA        -18        6.30      M   47  
4        No        53         NORWAY       -130       49.40      F   48  
...     ...       ...            ...        ...         ...    ...  ...  
18214    No       111          JAPAN         35       12.25      M   31  
18215    No        58          CHINA         40       14.00      F   40  
18216    No         2       MALAYSIA         18        6.30      M   57  
18217    No         3       VIET NAM         18        6.30      M   63  
18218    No        22      HONG KONG         26        9.10      F   35  

[18219 rows x 12 columns]

1. Tampilkan tabel sampai 5 baris pertama

In [178]:
cur = db_con.cursor()
q1= '''
SELECT * FROM travel_insurance LIMIT 5
'''

cur.execute(q1)
result = cur.fetchall()

In [179]:
pd.read_sql_query(q1,db_con)

c:\Users\praba\anaconda3\lib\site-packages\pandas\io\sql.py:761: UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
  warnings.warn(


id agency    agency_type distribution_channel        product_name claim  \
0   1    CBH  Travel Agency              Offline  Comprehensive Plan    No   
1   2    CBH  Travel Agency              Offline  Comprehensive Plan    No   
2   3    JZI       Airlines               Online          Value Plan    No   
3   4    JZI       Airlines               Online          Basic Plan    No   
4   5    KML  Travel Agency               Online        Premier Plan    No   

   duration    destination  net_sales  commission gender  age  
0       186       MALAYSIA        -29        9.57      F   81  
1       186       MALAYSIA        -29        9.57      F   71  
2        66  UNITED STATES       -121       42.35      F   44  
3         1       MALAYSIA        -18        6.30      M   47  
4        53         NORWAY       -130       49.40      F   48

2. Berasal dari agensi travel mana saja nasabah yang klaim asuransi?

In [180]:
q2= '''
SELECT DISTINCT agency FROM travel_insurance
'''

cur.execute(q2)
result = cur.fetchall()

In [181]:
pd.read_sql_query(q2,db_con)

c:\Users\praba\anaconda3\lib\site-packages\pandas\io\sql.py:761: UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
  warnings.warn(


agency
0     CBH
1     JZI
2     KML
3     C2B
4     JWT
5     RAB
6     ART
7     CSR
8     CCR
9     ADM
10    CWT
11    LWC
12    TTW
13    TST

Jika kita mengabaikan status klaim baik diterima ataupun tidak, terdapat 13 agensi travel yang menawarkan asuransi perjalanan dari tipe apapun.

3. Tipe agensi apa yang paling banyak yang klaim asuransinya diterima?

In [182]:
q3= '''
SELECT agency_type, count(claim) as total FROM 
	(SELECT * FROM travel_insurance WHERE claim = "Yes") as q1
GROUP BY agency_type
'''

cur.execute(q3)
result = cur.fetchall()

In [183]:
pd.read_sql_query(q3,db_con)

c:\Users\praba\anaconda3\lib\site-packages\pandas\io\sql.py:761: UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
  warnings.warn(


agency_type  total
0       Airlines    584
1  Travel Agency     55

Dapat dilihat bahwa jenis agensi yang berhasil untuk mengklaim asuransinya kebanyakan adalah tipe airlines yang jumlahnya cukup timpang dibandingkan dengan travel agency. Hal ini dapat berarti bahwa asuransi yang ditawarkan oleh travel agency banyak yang tidak dapat di klaim.

4. Destinasi negara mana yang paling banyak diterima klaim asuransinya?

In [184]:
q4= '''
SELECT destination, count(claim) as total FROM 
	(SELECT * FROM travel_insurance WHERE claim = "Yes") as q1
GROUP BY destination ORDER BY total DESC LIMIT 10
'''

cur.execute(q4)
result = cur.fetchall()

In [185]:
pd.read_sql_query(q4,db_con)

c:\Users\praba\anaconda3\lib\site-packages\pandas\io\sql.py:761: UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
  warnings.warn(


destination  total
0           SINGAPORE    550
1       UNITED STATES     16
2               CHINA     12
3           INDONESIA     10
4            MALAYSIA      8
5            THAILAND      6
6               JAPAN      5
7           HONG KONG      5
8  KOREA, REPUBLIC OF      5
9            VIET NAM      4

Perjalanan travel yang banyak sukses dalam mengklaim asuransinya di kasus ini adalah Singapur degan jumlah 550 klaim. Hal ini dapat disebabkan oleh banyaknya orang yang bepergian ke Singapur, namun juga dapat karena permasalahan lain yang banyak dialami bersama oleh user yang bepergian ke destinasi tersebut.

5. Berapa nilai minimum, maksimum, dan rata-rata usia yang pengajuan klaim asuransinya diterima?

In [186]:
q5 = '''
SELECT avg(age) AS mean,
min(age) as minimum,
max(age) as maximum FROM
	(SELECT age FROM travel_insurance WHERE claim = "Yes") as keep
'''

cur.execute(q5)
result = cur.fetchall()

In [187]:
pd.read_sql_query(q5,db_con)

c:\Users\praba\anaconda3\lib\site-packages\pandas\io\sql.py:761: UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
  warnings.warn(


mean  minimum  maximum
0  38.8294       14      118

Dapat dilihat bahwa individu yang berhasil untuk mengklaim asuransinya adalah individu dengan umur sekitar 38. Adapun individu termuda dan tertua yang berhasil untuk mengklaim asuransinya adalah 14 dan 118 tahun. Dari data ini, dapat didalami lagi masing masing alasan dibalik mengapa mereka dapat mengklaim asuransi. Walaupun begitu, para agensi travel dapat berfokus kepada kebutuhan/permaslahan perjalanan pada individu yang berusia akhir 30 an.

6. Jika agen asuransi hanya mengurusi klaim yang tujuan negaranya Singapura, Thailand, dan Malaysia, berapa total komisi yang didapat dari klaim yang diterima? (anggap komisi dalam USD)

In [188]:
q6 = '''
SELECT sum(commission) as total FROM 
	(SELECT * FROM travel_insurance
    WHERE destination = "SINGAPORE" 
    OR destination = "THAILAND" 
    OR destination = "MALAYSIA") as countries

'''

cur.execute(q6)
result = cur.fetchall()

In [189]:
pd.read_sql_query(q6,db_con)

c:\Users\praba\anaconda3\lib\site-packages\pandas\io\sql.py:761: UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
  warnings.warn(


total
0  193914.34

Dari total komisi dengan tujuan Singapur, Thailand, dan Malaysia yang sebelumnya memiliki nilai rendah pada jumlah klaim yang berhasil, agensi travel mendapatkan $193.914,34. Informasi tersebut dapat digunakan jika mereka ingin menganalisa lebih lanjut terhadap komisi yang akan didapatkan jika berfokus kepada tujuan destinasi lain, dan untuk menganalisa lebih lanjut terhadap pengalaman customer yang menggunakan servis asuransi tersebut.

## Conclusions

**Bahasa Indonesia**

Dari analisis yang dilakukan di atas dapat disimpulkan bahwa meskipun terdapat 13 agen yang menyediakan jasa asuransi perjalanan, namun jenis agen yang paling banyak mendapatkan klaim asuransinya adalah melalui maskapai penerbangan (584). Ditemukan pula bahwa negara tujuan yang paling banyak diklaim berhasil diklaim adalah Singapura, disusul Amerika Serikat. Usia pengguna yang biasanya bisa mendapatkan klaim adalah sekitar 38 tahun. Artinya, jika mereka ingin mengoptimalkan layanan mereka dalam asuransi perjalanan, mereka dapat menganalisis kebutuhan orang-orang yang akan pergi ke Singapura dan AS, dan berusia akhir tiga puluhan. Mereka juga dapat mempertimbangkan untuk berfokus pada penawaran layanan mereka melalui maskapai penerbangan.

Selain itu, ditemukan juga jika mereka hanya mengurus klaim dengan tujuan Singapura, Thailand, dan Malaysia, mereka mendapatkan $193.914,34. Untuk analisis selanjutnya, mereka dapat mencoba menghitung berapa total komisi yang akan mereka dapatkan jika fokus pada destinasi teratas dengan klaim terbanyak.


**English**

From the analysis made above, it can be inferred that although there are 13 agencies that provide travel insurance services, the agency type that gets their insurance successfully claimed the most is through airlines (584). It's also found that the country destination that gets successfully claimed the most is to Singapore, followed by The US. The user's age that is typically able to get claims is around 38 years old. This means that if they want to optimize their services in travel insurance, they could analyze the needs of people who is going to Singapore and the US, and is in their late thirties. They could also consider focusing on offering their services through airlines instead.

Other than that, it's also found that if they only take care of the claims with the destination of Singapore, Thailand, and Malaysia, they get $193.914,34. For the next analysis, they could try to calculate how much total commissions that they would get if they focused on the top destinations with the most claims.